core/controllers/certificate_assessment_test.py

```
@@ -427,109 +455,360 @@ def test_get_returns_stub_offerings_for_classroom(self) -> None:
+                handler.InvalidInputException, 'invalid attempt'
+            ):
+                handler.post('attempt_1')
+
+    def test_post_accepts_non_string_selected_answer(self) -> None:
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        empty_topic_versions: Dict[str, int] = {}
+        empty_question_versions: Dict[str, int] = {}
+        empty_question_topic_links: Dict[str, List[str]] = {}
+        version_data: Dict[
+            str, Union[str, int, Dict[str, int], Dict[str, List[str]]]
```

Ruff (0.16.1)

[warning] Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)